src/networking/BodyService.ts

```diff
@@ -11,6 +11,11 @@ export default class BodyService extends ModelService {
     super(COLLECTION_NAME.Body, firebaseConfig);
   }
 
+  async getBodyByName(name: string): Promise<Body> {
+    const networkQueryResponse = this.networkService.getDocument(name, COLLECTION_NAME.Body);
+    return this.createModel(networkQueryResponse, Body, `getBodyByName(${name})`);
```

Comment:
The call to .getDocument won't work as intended because getDocument requires a document id as the first arg?

Could you create a documentRef from `name` field?
`const documentRef = doc(NetworkService.db, collectionName, name);`

Reply:
Hmm, if `name` is not the primary key of documents in the body collection, I think I will have to make a query and return the top result.  I was thinking that it was the PK.